From 39d7ce2f157d7a33f1333788d4ff5c1238a12450 Mon Sep 17 00:00:00 2001
From: Kevin McCarthy <kevin@8t8.us>
Date: Sat, 4 Jun 2016 11:32:03 -0700
Subject: [PATCH 29/37] Sidebar clean up: building and drawing logic.

Fix the autoconf/makefile.am changes to be consistent.

Create a global SidebarNeedsRedraw to indicate a redraw is needed,
instead of putting sb_draw() everywhere in the code.

Create a menu_redraw_sidebar() function and use the REDRAW_SIDEBAR
flag instead of piggy-backing it inside the index loop.

Fix curs_main.c and pager.c to be a bit cleaner by using the global and
REDRAW_SIDEBAR.

Start to clean up some of the buffy code, but this needs to refactored
and fixed.
---
 addrbook.c | 2 +-
 compress.c | 2 +-
 globals.h  | 2 +-
 newsrc.c   | 2 +-
 sendlib.c  | 2 +-
 5 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/addrbook.c b/addrbook.c
index 8ede157..df0a481 100644
--- a/addrbook.c
+++ b/addrbook.c
@@ -81,7 +81,7 @@ alias_format_str (char *dest, size_t destlen, size_t col, int cols, char op, con
 
 static void alias_entry (char *s, size_t slen, MUTTMENU *m, int num)
 {
-  mutt_FormatString (s, slen, 0, COLS - SidebarWidth, NONULL (AliasFmt), alias_format_str, (unsigned long) ((ALIAS **) m->data)[num], MUTT_FORMAT_ARROWCURSOR);
+  mutt_FormatString (s, slen, 0, MuttIndexWindow->cols, NONULL (AliasFmt), alias_format_str, (unsigned long) ((ALIAS **) m->data)[num], MUTT_FORMAT_ARROWCURSOR);
 }
 
 static int alias_tag (MUTTMENU *menu, int n, int m)
diff --git a/compress.c b/compress.c
index 6448924..4c23b5c 100644
--- a/compress.c
+++ b/compress.c
@@ -379,7 +379,7 @@ get_compression_cmd (const CONTEXT *ctx, const char *cmd)
 
 	char expanded[_POSIX_PATH_MAX];
 
-	mutt_FormatString (expanded, sizeof (expanded), 0, COLS - SidebarWidth, cmd, cb_format_str, (unsigned long) ctx, 0);
+	mutt_FormatString (expanded, sizeof (expanded), 0, MuttIndexWindow->cols, cmd, cb_format_str, (unsigned long) ctx, 0);
 	return safe_strdup (expanded);
 }
 
diff --git a/globals.h b/globals.h
index 336fa64..0b33e81 100644
--- a/globals.h
+++ b/globals.h
@@ -244,9 +244,9 @@ WHERE short ScoreThresholdDelete;
 WHERE short ScoreThresholdRead;
 WHERE short ScoreThresholdFlag;
 
-WHERE short SidebarWidth INITVAL(0);
 #ifdef USE_SIDEBAR
 WHERE short SidebarRefreshTime;
+WHERE short SidebarWidth INITVAL(0);
 WHERE LIST *SidebarWhitelist INITVAL(0);
 WHERE int SidebarNeedsRedraw INITVAL (0);
 #endif
diff --git a/newsrc.c b/newsrc.c
index 9072219..687b726 100644
--- a/newsrc.c
+++ b/newsrc.c
@@ -973,7 +973,7 @@ NNTP_SERVER *nntp_select_server (char *server, int leave_lock)
   /* load .newsrc */
   if (rc >= 0)
   {
-    mutt_FormatString (file, sizeof (file), 0, COLS - SidebarWidth, NONULL (NewsRc),
+    mutt_FormatString (file, sizeof (file), 0, MuttIndexWindow->cols, NONULL (NewsRc),
 		       nntp_format_str, (unsigned long)nserv, 0);
     mutt_expand_path (file, sizeof (file));
     nserv->newsrc_file = safe_strdup (file);
diff --git a/sendlib.c b/sendlib.c
index 4b3f1c2..4350f1c 100644
--- a/sendlib.c
+++ b/sendlib.c
@@ -2393,7 +2393,7 @@ mutt_invoke_sendmail (ADDRESS *from,	/* the sender */
   {
     char cmd[LONG_STRING];
 
-    mutt_FormatString (cmd, sizeof (cmd), 0, COLS - SidebarWidth, NONULL (Inews), nntp_format_str, 0, 0);
+    mutt_FormatString (cmd, sizeof (cmd), 0, MuttIndexWindow->cols, NONULL (Inews), nntp_format_str, 0, 0);
     if (!*cmd)
     {
       i = nntp_post (msg);
-- 
2.9.0

